pkg/agent/util/ndp/ndp_linux.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ndp
+
+import (
+	"fmt"
+	"net"
+
+	"golang.org/x/net/icmp"
+	"golang.org/x/net/ipv6"
+)
+
+const (
+	// Option Length, 8-bit unsigned integer. The length of the option (including the type and length fields) in units of 8 octets.
+	// The value 0 is invalid. Nodes MUST silently discard an ND packet that contains an option with length zero.
+	// https://datatracker.ietf.org/doc/html/rfc4861
+	ndpOptionLen = 1
+
+	// ndpOptionType
+	// 	Option Name                             Type
+	//
+	// Source Link-Layer Address                    1
+	// Target Link-Layer Address                    2
+	// Prefix Information                           3
+	// Redirected Header                            4
+	// MTU                                          5
+	ndpOptionType = 2
+
+	// Minimum byte length values for each type of valid Message.
+	naLen = 20
+
+	// expected IPv6 hop limit, hop limit is always 255 for all NDP messages, RFC 4861.
+	hopLimit = 255
+)
+
+func checkIPv6(ip net.IP) error {
+	if ip.To16() == nil || ip.To4() != nil {
+		return fmt.Errorf("invalid IPv6 address: %s", ip.String())
+	}
+	return nil
+}
+
+// NeighborAdvertisement sends an NDP Neighbor Advertisement over interface 'iface' from 'srcIP'.
+func NeighborAdvertisement(srcIP net.IP, iface *net.Interface) error {
+	if err := checkIPv6(srcIP); err != nil {
+		return err
+	}
+
+	addrs, err := iface.Addrs()
+	if err != nil {
+		return fmt.Errorf("interface address error: %v", err)
+	}
+
+	ipAddr := &net.IPAddr{}
```

Comment:
```suggestion
	var ipAddr *net.IPAddr
```
Since you create a new instance in the loop.

Reply:
done